src/components/Board/Board.reducer.js

```diff
@@ -133,6 +146,37 @@ function boardReducer(state = initialState, action) {
         ...state,
         output: [...action.output]
       };
+    case TOGGLE_SELECT:
+      return {
+        ...state,
+        tileSelectable: !state.tileSelectable
+      };
+    case SELECT_TILE:
+      return {
+        ...state,
+        selectedTileIds: [...state.selectedTileIds, action.tileId]
+      };
+    case DESELECT_TILE:
+      return {
+        ...state,
+        selectedTileIds: [
+          ...state.selectedTileIds.filter(tileId => tileId !== action.tileId)
```

Comment:
We might not need [ ...filter ] because filter returns a new array

Reply:
good catch!